src/test/java/com/thirdcc/webapp/web/rest/EventActivityResourceIT.java

```diff
@@ -275,25 +448,217 @@ public void updateNonExistingEventActivity() throws Exception {
+            .content(TestUtil.convertObjectToJsonBytes(eventActivityDTO)))
+            .andExpect(status().isBadRequest());
+
+        List<EventActivity> eventActivityList = eventActivityRepository.findAll();
+        assertThat(eventActivityList).hasSize(databaseSizeBeforeUpdate);
+    }
+
+    @Test
+    public void updateEventActivity_WithEventCancelled_ShouldThrow400() throws Exception {
+        event.setStatus(EventStatus.CANCELLED);
+        Event savedEvent = initEventDB();
+        EventActivity savedEventActivity = initEventActivityDB(savedEvent);
+
+        int databaseSizeBeforeUpdate = eventActivityRepository.findAll().size();
+
+        EventActivityDTO eventActivityDTO = createUpdateEventActivityDTO();
+        eventActivityDTO.setId(savedEventActivity.getId());
+
+        restEventActivityMockMvc.perform(put("/api/event-activities")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(eventActivityDTO)))
+            .andExpect(status().isBadRequest());
+
+        List<EventActivity> eventActivityList = eventActivityRepository.findAll();
+        assertThat(eventActivityList).hasSize(databaseSizeBeforeUpdate);
+    }
+
+    @Test
+    public void updateEventActivity_WithEventEarlierThanToday_ShouldThrow400() throws Exception {
+        Event savedEvent = initEventDB();
+        EventActivity savedEventActivity = initEventActivityDB(savedEvent);
+
+        int databaseSizeBeforeUpdate = eventActivityRepository.findAll().size();
+
+        EventActivityDTO eventActivityDTO = createUpdateEventActivityDTO();
+        eventActivityDTO.setId(savedEventActivity.getId());
+        eventActivityDTO.setStartDate(Instant.now().minus(1, ChronoUnit.SECONDS));
+
+        assertThat(eventActivityDTO.getStartDate()).isBefore(Instant.now());
+
+        restEventActivityMockMvc.perform(put("/api/event-activities")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(eventActivityDTO)))
+            .andExpect(status().isBadRequest());
+
+        List<EventActivity> eventActivityList = eventActivityRepository.findAll();
+        assertThat(eventActivityList).hasSize(databaseSizeBeforeUpdate);
+    }
+
+    @Test
+    public void updateEventActivity_WithStartDateEarlierThanEventStartDate_ShouldThrow400() throws Exception {
+        event.setStartDate(Instant.now().plus(1, ChronoUnit.DAYS));
+        Event savedEvent = initEventDB();
+        EventActivity savedEventActivity = initEventActivityDB(savedEvent);
+
+        int databaseSizeBeforeCreate = eventActivityRepository.findAll().size();
+
+        EventActivityDTO eventActivityDTO = createUpdateEventActivityDTO();
+        eventActivityDTO.setId(savedEventActivity.getId());
+        eventActivityDTO.setStartDate(savedEvent.getStartDate().minus(1, ChronoUnit.SECONDS));
+
+        assertThat(savedEvent.getStartDate()).isAfter(Instant.now()); //for more accurate testing
```

Comment:
same with the comment mention above
why appear on one test case only

Reply:
the original motivation is to ensure line 97 of EventActivityServiceImpl.java
`if (eventActivityDTO.getStartDate().isBefore(event.getStartDate()))`
 is tested.

But line 94 of EventActivityServiceImpl.java 
`if (eventActivityDTO.getStartDate().isBefore(Instant.now()))`
may return true, and proceed to throw error

so this line (line 544 EventActivityResourceIT.java) was added
`assertThat(savedEvent.getStartDate()).isAfter(Instant.now()); //for more accurate testing`
to ensure that we are really testing line 97 of EventActivityServiceImpl.java

**But** this test case is removed in f8fd7839e49f49048eb7115c741ac59c12452a2c. Due to a business logic change. Refer #8